src/tally.cpp

```diff
@@ -440,14 +466,78 @@ std::string pyne::Tally::mcnp(int tally_index, std::string mcnp_version) {
     } else if (tally_type.find("Current") != std::string::npos) {
       // makes no sense in mcnp
     }
+  
+  
+  } else if (entity_type.find("Mesh") != std::string::npos) {
+    output << "FMESH4:" << particle_token << "  ";
+    output << "GEOM=";
+    std::stringstream sup_var;
+
+    if (entity_geometry.find("XYZ") != std::string::npos) {
+      output << "XYZ ";
+    } else if (entity_geometry.find("Cylinder") != std::string::npos) {
+      output << "CYL ";
+      if (axl[0] != 0 || axl[1] != 0 || axl[2] != 0) {
+        sup_var << "AXL=";
+        for (int i = 0; i < 3; i++) {
+              sup_var << axl[i] << " ";
+        }
+      }
+      if (vec[0] != 0 || vec[1] != 0 || vec[2] != 0) {
+        sup_var << "VEC=";
+        for (int i = 0; i < 3; i++) {
+              sup_var << vec[i] << " ";
+        }
+      }
+    }
+    output << " ORIGIN= " << origin[0] << " " << origin[1] << " " << origin[2]
+        << "\n";
+    std::string dir_name[3] = {"I", "J", "K"};
+    std::vector<double> meshes[3] = {i_meshs, j_meshs, k_meshs};
+    std::vector<int> bins[3] = {i_bins, j_bins, k_bins};
+
+    for (int j = 0; j < 3; j++) {
+      output << "          ";
+      output << dir_name[j] << "MESH=";
+      for (int i = 0; i < meshes[j].size(); i++) {
```

Comment:
as above, a method to convert vectors of arbitrary length into a string "a b c d ..."

Reply:
maybe this is essentially the same as above...